syntax = "proto3";

message Point{
	float x = 1;
	float y = 2;
}
message Rectangle{
	Point point1 = 1;
	Point point2 = 2;
}
message Debug_Robot{
	Point pos = 1;
	float dir = 2;
}
message Debug_Line {
	Point start = 1;
	Point end = 2;
	bool FORWARD = 3;
	bool BACK = 4;
}
message Debug_Arc {
	Rectangle rectangle = 1;
	float start = 2;
	float end = 3;
	bool FILL = 4;
}
message Debug_Polygon {
	repeated Point vertex = 1;
	bool FILL = 2;
}
message Debug_Text {
	Point pos = 1;
	string text = 2;
}
message Debug_Curve_ {
	float num = 1;
	float maxLimit = 2;
	float minLimit = 3;
}
message Debug_Curve {
	Point start = 1;
	Point p1 = 2;
	Point p2 = 3;
	Point end = 4;
}
message Debug_Points {
	repeated Point point = 1;
}
message Debug_Msg {
	enum Debug_Type {
		ARC 	= 0;
		LINE 	= 1;		
		TEXT 	= 2;
		ROBOT 	= 3;
		CURVE 	= 4;
		POLYGON = 5;
		Points   = 6;
	}
	Debug_Type type = 1;
	enum Color {
		WHITE = 	0;
		RED = 		1;
		ORANGE = 	2;
		YELLOW = 	3;
		GREEN = 	4;
		CYAN = 		5;
		BLUE = 		6;
		PURPLE = 	7;
		GRAY = 		8;
		BLACK = 	9;
	}
	Color color = 2;
	// one of the follow >>> need to be tested with TYPE <oneof> 
	Debug_Arc arc = 	3;
	Debug_Line line = 	4;
	Debug_Text text = 	5;
	Debug_Robot robot = 	6;
	Debug_Curve_ curve = 	7;
	Debug_Polygon polygon = 8;
	Debug_Points points = 9;
}
message Debug_Msgs{
	repeated Debug_Msg msgs= 1;
}
